apps/mobile/src/components/widgets/tokens/tokens-widget.tsx

```diff
@@ -1,38 +1,41 @@
-import React from 'react';
+import React, { ReactNode } from 'react';
 
+import { AccountId } from '@/models/domain.model';
+import {
+  useBitcoinAccountTotalBitcoinBalance,
+  useWalletTotalBitcoinBalance,
+} from '@/queries/balance/bitcoin-balance.query';
+import { HasChildren } from '@/utils/types';
 import { t } from '@lingui/macro';
 
 import { Money } from '@leather.io/models';
-import { Flag, ItemLayout } from '@leather.io/ui/native';
+import { BtcAvatarIcon, Flag, ItemLayout } from '@leather.io/ui/native';
 
-import { type Token } from '../../../mocks/tokens.mocks';
+import { getMockTokens } from '../../../mocks/tokens.mocks';
 import { Balance } from '../../balance/balance';
 import { Widget, WidgetHeader } from '../components/widget';
 
-interface TokensWidgetProps {
-  tokens: Token[];
-  totalBalance: Money;
-}
-
 function showChain(chain: string) {
   if (chain === 'Stacks blockchain' || chain === 'Bitcoin blockchain') return;
   return chain;
 }
 
-export function TokensWidget({ tokens, totalBalance }: TokensWidgetProps) {
+export function TokensWidget({ children }: HasChildren) {
   return (
     <Widget
-      header={
-        <WidgetHeader
-          title={t({
-            id: 'tokens.header_title',
-            message: 'My tokens',
-          })}
-          totalBalance={totalBalance}
-        />
-      }
+      header={<WidgetHeader title={t({ id: 'tokens.header_title', message: 'My tokens' })} />}
     >
-      {tokens.map(
+      {children}
+    </Widget>
+  );
+}
+
+export function AllAccountBalances() {
```

Comment:
Should we create a new folder structure for `balances`? I guess we should organise this into smaller components